src/MakerTemplateRendererInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Symfony\Bundle\MakerBundle;
+
+/**
+ *
+ * @author Jonathan Kablan <jonathan.kablan@gmail.com>
+ */
+interface MakerTemplateRendererInterface
+{
+    /**
+     * @param string $templateName
+     * @return string
+     */
+    public function supports(string $templateName): string;
```

Comment:
```suggestion
    public function supports(string $templateName): bool;
```

This will return a boolean. The idea is that `MakerRenderer` will look a bit different than you have. It will look like this:

```php
class MakerRenderer
{
    /**
     * @var MakerTemplateRendererInterface[]
     */
    private $templateRenderers;

    public function __construct(array $templateRenderers)
    {
        $this->templateRenderers = $templateRenderers;
    }

    public function renderTemplate(string $templateName, array $variables)
    {
        foreach ($this->templateRenderers as $templateRenderer) {
            if ($templateRenderer->supports($templateName)) {
                return $templateRenderer->render($templateName, $variables);
            }
        }
        
        throw new \Exception(sprintf('No template renderers for template "%s"', $templateName);
    }
}
```

There would be ONE class in MakerBundle that implements `MakerTemplateRendererInterface` (and users could add more by using a dependency injection tag). It would probably look like this:

```php
class MakerDefaultTemplateRenderer implements MakerTemplateRendererInterface
{
    public function supports(string $templateName): string
    {
        $templatePath = __DIR__.'/Resources/skeleton/'.$templateName;

        return file_exists($templatePath);
    }

    public function render(string $templateName, array $variables)
    {
        $templatePath = __DIR__.'/Resources/skeleton/'.$templateName;

        // this is basically the current logic for rendering templates
        $contents = $this->fileManager->parseTemplate($templatePath, $variables);
        $this->fileManager->dumpFile($templatePath, $contents);
    }
}
```